train.lua

```diff
@@ -11,43 +11,74 @@ local model_utils = require 'util.model_utils'
 local Comment2VoteSGDLoader = require 'util.Comment2VoteSGDLoader'
 local LSTM = require 'LSTM'
 
-rnn_size = 5
+rnn_size = 32
 num_layers = 1
 grad_clip = 5
 learning_rate = 2e-3
 learning_rate_decay = .97
 decay_rate = .95
-learning_rate_decay_after = 10
+learning_rate_decay_after = 10000
 batch_size = 128
 
 local split_sizes = {1, 0, 0} 
-data_dir = 'data/nba2'
+--data_dir = 'data/nba2'
+data_dir = 'data/test'
 local loader = Comment2VoteSGDLoader.create(data_dir, batch_size, split_sizes)
 vocab_size = loader.vocab_size
 
+gpuid = 1
+if gpuid >=0 then
+  local ok, cunn = pcall(require, 'cunn')
+  assert(ok, 'package cunn not found!')
+  local ok, cutorch = pcall(require, 'cutorch')
+  cutorch.setDevice(gpuid+1) 
+end
+
 encode, decode = {}, {}
 encode.rnn = LSTM.create(vocab_size, vocab_size, rnn_size, num_layers, 0)
 decode.rnn = LSTM.create(1, 1, rnn_size, num_layers, 1)
 decode.criterion = nn.MSECriterion()
 
+if gpuid >=0 then
+    for k,v in pairs(encode) do v:cuda() end
+    for k,v in pairs(decode) do v:cuda() end
+end
+
 params, grad_params = model_utils.combine_all_parameters(encode.rnn, decode.rnn)
 params:uniform(-.08, .08)
 
+
 dclone = {}
 for name,proto in pairs(decode) do
 	dclone[name] = model_utils.clone_many_times(proto, 1, not proto.parameters)
 end
 
-local init_state = {}
+eclone = {}
+local max_enc_len = 128
+for name, proto in pairs(encode) do
+  eclone[name] = model_utils.clone_many_times(proto, max_enc_len, not proto.parameters)
+end
+
+local init_state_enc = {}
+local init_state_dec = {}
 for i=1, num_layers do
 	local h_init = torch.zeros(batch_size, rnn_size)
-	table.insert(init_state, h_init:clone())
-	table.insert(init_state, h_init:clone())
+	if gpuid >=0 then h_init = h_init:cuda() end
+	table.insert(init_state_enc, h_init:clone())
+	table.insert(init_state_enc, h_init:clone())
+	table.insert(init_state_dec, h_init:clone())
```

Comment:
I have an issue with this. Should the hidden states being passed into the decoder be unbiased or should they be the hidden state outputs of the encoder? I would think the latter works better?